# General configure
tf_expiry: 0.3                # [s]
dock_timeout: 240             # [s]
slider_timeout: 15            # [s] - Waiting time for slider motor reachs max position
controller_frequency: 20.0    # [hz]
gear_ratio: 10.33
encoder_resolution: 82640.0
wheel_radius: 0.0625
back_laser_offset: 0.52322    # Distance from base_link to back_laser_link when slider at original position
steer_distance_threshold: 0.63

# Frames
base_link: base_link
charger_frame: charger_frame
first_frame: cart_frame
last_frame: last_frame
tag_frame: tag_frame
parallel_frame: parallel_frame

# Custom dock
# custom_dock_name:
#   clr55_dropoff:
#     action1:
#       rotate: -92
#       move: -0.5
#     action2:
#       rotate: 92

# Goal threshold
stop_trans_diff: 0.02         # [m]
y_tolerance_pid: 0.015        # [m]
max_parallel_offset: 0.02     # [m] - will move to parallel.c if exceeded
yaw_predock_tolerance: 0.055  # [rad]
stop_yaw_diff: 0.05           # [rad]
predock_tf_samples: 5         # tf samples to avg, parallel.c validation

# Velocity informations
max_angular_vel: 0.3                  # [rad/s]
min_angular_vel: 0.05                 # [rad/s]
max_linear_vel: 0.3
min_linear_vel: 0.035                 # [m/s]
# max_linear_vel: 0.1
# min_linear_vel: 0.035                 # [m/s]
rotate_to_heading_angular_vel: 0.5
max_angular_accel: 3.2
max_angular_deccel: 0.3
angle_threshold: 0.35
max_linear_vel_predock: 0.15      
max_x_lastmile: 0.03                  # [m/s] - Max linear velocity in lastmile state
max_x_out_dock: 0.2                   # [m/s] - Max linear velocity when go out dock
max_x_correction: 0.05                # [m/s] - Max linear velocity when use correct_robot function
max_x_with_retry_high_current: 0.1    # [m/s] - Max linear velocity when move with encoder
max_x_pid_steer: 0.035                # [m/s] - Max pid's linear velocity when use pid controller in steer state
min_x_pid_steer: 0.02                 # [m/s] - Min pid's linear velocity when use pid controller in steer state
max_x_pid_lastmile: 0.035             # [m/s] - Max pid's linear velocity when use pid controller in lastmile state
min_x_pid_lastmile: 0.02              # [m/s] - Min pid's linear velocity when use pid controller in lastmile state
max_x_go_out_dock: 0.2                # [m/s] - Max linear velocity when go out dock
max_x_custom_rotation: 0.1            # [m/s] - Max linear velocity when do custom rotation line 56
max_w_rotate_to_dock: 0.18            # [rad/s] - Max angular velocity when rotate with odom
max_w_correction: 0.12                # [rad/s] - Max angular velocity when use correct_robot function
max_w_predock: 0.09                   # [rad/s] - Max angular velocity when use do_predock function
max_w_custom_rotation: 0.2            # [rad/s] - Max angular velocity when do custom rotation line 56

k_p: 0.8
k_i: 0.0
k_d: 0.01
k_p_steer: 2.5
k_i_steer: 0.0
k_d_steer: 0.01

retry_count: 5                        # How many times to retry
debug_mode: True                      # Print messages for debugging.